### PR TITLE
libdispatch: update to 5.10

### PR DESCRIPTION
**Pull request by ognevny**
patch **is not** included in the new release https://github.com/apple/swift-corelibs-libdispatch/compare/swift-5.9.2-RELEASE...swift-5.10-RELEASE

### CODE REVIEW

**Comment by ognevny** [comment]
libdispatch's `libdispatch.dll` is linked against `BlocksRuntime.dll`, now `libBlocksRuntime.dll`

**Comment by ognevny** [comment]
> libdispatch's `libdispatch.dll` is linked against `BlocksRuntime.dll`, not `libBlocksRuntime.dll`

I've decided to rename dll so it's `BlocksRuntime.dll` now. is this ok?

**Comment by mmuetzel** [comment]
> I've decided to rename dll so it's `BlocksRuntime.dll` now. is this ok?

Won't that break the link between the import library and the dll?

**Comment by ognevny** [comment]
maybe we should change this line https://github.com/apple/swift-corelibs-libdispatch/blob/main/dispatch%2Fgeneric%2Fmodule.modulemap#L5 ? I can try it only tomorrow

File: mingw-w64-libdispatch/PKGBUILD [hunk 1]
```diff
@@ -35,47 +40,47 @@ build() {
...
+           "${MINGW_PACKAGE_PREFIX}-libwinpthread")
   cd "${srcdir}/build-${MSYSTEM}"
 
   DESTDIR="${pkgdir}" "${MINGW_PREFIX}"/bin/cmake.exe --install .
-  rm "${pkgdir}${MINGW_PREFIX}/bin/libBlocksRuntime.dll"
-  rm "${pkgdir}${MINGW_PREFIX}/lib/libBlocksRuntime.dll.a"
-  rm "${pkgdir}${MINGW_PREFIX}/include/Block.h"
+
+  mkdir -p dest${MINGW_PREFIX}/{bin,lib,include}
+  mv "${pkgdir}${MINGW_PREFIX}/bin/libBlocksRuntime.dll" "dest${MINGW_PREFIX}/bin/BlocksRuntime.dll"
```
Comment (mmuetzel) [inline mingw-w64-libdispatch/PKGBUILD:68]: The dll should probably not be renamed when packaging.

Comment (ognevny) [reply mingw-w64-libdispatch/PKGBUILD:68]: I've found that it's enforced to link `BlocksRuntime` library. I'm trying to patch it now

Comment (MehdiChinoune) [reply mingw-w64-libdispatch/PKGBUILD:68]: You know there is difference between dynamic library and import library.

**Comment by ognevny** [comment]
now I recieve this errors
```
swift-corelibs-libdispatch-swift-5.10-RELEASE/src/BlocksRuntime/data.c:27:18: error: definition of dllimport data
   27 | BLOCK_ABI void * _NSConcreteMallocBlock[32] = { 0 };
      |                  ^
swift-corelibs-libdispatch-swift-5.10-RELEASE/src/BlocksRuntime/data.c:28:18: error: definition of dllimport data
   28 | BLOCK_ABI void * _NSConcreteAutoBlock[32] = { 0 };
      |                  ^
swift-corelibs-libdispatch-swift-5.10-RELEASE/src/BlocksRuntime/data.c:29:18: error: definition of dllimport data
   29 | BLOCK_ABI void * _NSConcreteFinalizingBlock[32] = { 0 };
      |                  ^
swift-corelibs-libdispatch-swift-5.10-RELEASE/src/BlocksRuntime/data.c:30:18: error: definition of dllimport data
   30 | BLOCK_ABI void * _NSConcreteWeakBlockVariable[32] = { 0 };
      |                  ^
4 errors generated.
```

**Comment by MehdiChinoune** [comment]
```
/mingw64/bin/libdispatch.dll:
  	libgcc_s_seh-1.dll => not found
  	libwinpthread-1.dll => D:\M\msys64\mingw64\bin\libwinpthread-1.dll (0x0000025b18c00000)
  	BlocksRuntime.dll => D:\M\msys64\mingw64\bin\BlocksRuntime.dll (0x0000025b18c00000)
```
missing `gcc-libs` in depends for libdispatch.

**Comment by ognevny** [comment]
done

**Comment by ognevny** [comment]
should I send the patch upstream?

**Comment by MehdiChinoune** [comment]
> should I send the patch upstream?

That kind of patch will probably be rejected, MSVC doesn't prefix dynamic libraries, so they won't accepet because It will break MSVC build.

**Pull request merged**